refactor(editor): use always the provided language server

Sysix opened the pull request:
Now, the editor will not look into `node_modules` folder to use a custom `oxc_language_server`.
The editor will always use the provided language server.

## Reasons

Because the language server in `node_modules` could be a different version than the extension and changes like #10890 can break the communication between them.

Avoiding the JS Wrapper and resolving #9925 and avoiding a new setting #11018

closes #9925 
closes #11018

Sysix commented:
* **#11115** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11115?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11115?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11115).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
### Merge activity

* **May 18, 8:20 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(editor): use always the provided language server (#11115)

Now, the editor will not look into `node_modules` folder to use a custom `oxc_language_server`.
The editor will always use the provided language server.

## Reasons

Because the language server in `node_modules` could be a different version than the extension and changes like #10890 can break the communication between them.

Avoiding the JS Wrapper and resolving #9925 and avoiding a new setting #11018

closes #9925
closes #11018
